eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/OfferTypes.scala

```diff
@@ -153,17 +149,17 @@ object Offers {
     val quantityMin: Option[Long] = records.get[QuantityMin].map(_.min)
     val quantityMax: Option[Long] = records.get[QuantityMax].map(_.max)
 
-    val nodeIdXOnly: ByteVector32 = records.get[NodeId].get.xonly
+    val nodeId: PublicKey =
+      records.get[NodeId].map(_.publicKey).getOrElse(records.get[Paths].get.paths.head.blindedNodes.last.blindedPublicKey)
```

Comment:
I'm not sure about using a `node_id` from the first blinded path here.
What about making `node_id` required regardless of whether there is a blinded path?
This gives more flexibility to the offer creator, doesn't it? The blinded paths don't have to match the offer's `node_id` this way, they are a completely separate concern.

Reply:
In case the offer creator wants to use a blinded path, it doesn't make sense to require an additional `node_id` which would likely just be a random single-use key. It would just use QR code space, that's it.